### PR TITLE
ROP no longer fails to load on add reg to rsp/esp

### PR DESCRIPTION
**Pull request by johnandersen777**
ROP.__load takes any gadgets that add to esp and converts the amount
they add to esp to an int. However this threw an exception on finding
the instruction `add esp, eax` because eax is not an int.

Signed-off-by: John Andersen johnandersenpdx@gmail.com

### CODE REVIEW

**Comment by arthaud** [comment]
Hi @pdxjohnny,

I haven't looked at this code for a while. I am not sure ignoring silently the exception is a good idea. I will take a deeper look at this later this week and I'll keep you updated. Thank you!

**Comment by johnandersen777** [comment]
Great thanks! It would be great if we were to put the gadgets found to use somewhere :)

**Comment by johnandersen777** [comment]
I updated it so that its optional to load gadgets, which is helpful if you have a large binary and didn't want to search every time. It continues after it loads from the cache. This also solves the other problem where it crashed because it wont fail to load that `add %eax, %eax` if it doesn't do any searching.

File: pwnlib/rop/rop.py [hunk 1]
```diff
@@ -379,13 +379,11 @@ class ROP:
     #: which is not contiguous
     migrated = False
 
-    def __init__(self, elfs, base=None, **kwargs):
+    def __init__(self, elfs, base=None, should_load_gadgets=True, **kwargs):
         """
         Arguments:
             elfs(list): List of ``pwnlib.elf.ELF`` objects for mining
         """
-        import ropgadget
```
Comment (johnandersen777) [inline pwnlib/rop/rop.py:387]: Also it looked like this wasn't needed here so I removed it. Motivation being that capstone can be annoying to install.

**Comment by arthaud** [comment]
Hi @pdxjohnny,

Thank you for the PR.

**Pull request merged**